### DISCUSSION TITLE
Debugging and Setup Help

### DISCUSSION THREAD

**Discussion by carlaKC** [bitcoin-dev-project/sim-ln#124] [General]
If you're struggling to install, configure or run SimLN let us know what you're struggling with! 

Setups questions of any level are welcome and wanted.

**Comment by cndolo** [comment]
Hi. First-time user of `sim-ln` here and I'm glad to have found this tool as it seems to handle a lot of what I normally have to do. I have some questions I hope to get help on.

I have a private Lightning network with 5 nodes, A to E, each running on separate VMs. What I want to achieve is the following:

1. A sends random payments to the other nodes.
2. Random payments between all pairs of nodes.

`sim-ln` is running on the same machine as A.

AFAICS, 1. seems tricky as only A is provided in the sim file and [payments will only be sent between the nodes that are provided so that liquidity does not "drain" from the simulation](https://github.com/bitcoin-dev-project/sim-ln#setup---random-activity). How can A make random payments to any node based on their channel graph?

In order to achieve 2., `sim-ln` needs to have the access details for all nodes. Since they are all running on separate machines, machine A needs access to the other four in order to read the LN config files. Is there a better way of achieving random payments between all nodes?

**Reply by carlaKC** [reply]
> A sends random payments to the other nodes.
> Random payments between all pairs of nodes.

Just to make sure that I fully understand your goal here, do you want:
1. `A` to send random amount payments to `B`/`C`/`D`/`E` at random intervals
2. Random payments to flow between `A`/`B`/`C`/`D`/`E` (should A be included here?)

>How can A make random payments to any node based on their channel graph? 

If you include the other nodes in your simfile, then A will randomly select them as destinations (weighted by their channel capacity). Right now we don't support random destinations from your channel graph, but feel free to open up an issue (or a PR) if that's an important feature to you! You could possibly also take a look at specifying [defined activity](https://github.com/bitcoin-dev-project/sim-ln#setup---defined-activity) for A that allows you to specify the other nodes as destinations without having to have access to the other nodes. 

>  Since they are all running on separate machines, machine A needs access to the other four in order to read the LN config files.

Sadly there's no realy getting around needing access to the nodes that you want to send payments from (which I believe is what you want for 2). You're always going to need grpc access + macaroon/certs to be able to run sendpayment commands.
